### ISSUE TITLE
While waiting for vector support maybe let's use higher quality images for cardsets (with first new builds)?

### ISSUE THREAD

**Issue by jan-kleks** [shlomif/PySolFC#50]
Before creating PySolFC 2.1 builds for various systems, I suggest taking into account the fact that currently used cardsets are extremely small, and they look really, really bad on modern screens. To **temporarily** alleviate this issue and **to avoid negative user feedback** (while waiting for proper vector support), we might simply ship PySolFC with at least one additional (which could be used as the default one) higher resolution cardset. [I mean something like this guy did here](https://ubuntu-mate.community/t/pysolfc-cardset-for-1920x1080-screen/12308) (ignore the card backs s/he suggested, just focus on the cards). AFAIK, these seem to be cardsets which are in the public domain, we might need to ask him/her about this.

But PySolFC definitely looks much better this way... Some time ago I talked to shlomif on IRC that PySolFC needs a major graphical overhaul, taking care of (even if temporarily) the most serious issue, i.e. small cardsets could be a step in the right direction.

**Comment by jan-kleks** [comment]
A very preliminary version of the reddit/Hacker News announcement. Any suggestions are welcome, I’m trying to keep it short, so that people would not fall asleep…

Before making the announcement: 

Maybe we could set up a very, very simple website such as this one (created with https://pages.github.com): http://openhantek.org BTW, do we still need SourceForge? What about redirecting everything to GitHub to keep the development focused?

I guess we do not consider changing the name of the project yet as discussed here: https://github.com/shlomif/PySolFC/issues/13

As I mentioned many times before we could REALLY use at least one cardset which would be compatible with Full HD screens before “going vector”: https://github.com/shlomif/PySolFC/issues/50 We might even try resizing our very retro pixelated default PySolFC cardset for this (it shouldn’t look much worse…).


TITLE: PySolFC is back, new stable release! — open-source collection of +1000 solitaire card/mahjong games

PySolFC is a free and open-source collection of +1000 solitaire card and mahjong games with a great set of features and solver/analysis tools. The application is almost 20 years old [I will put this link here: http://solitaire.vegard2.net/pysol.html], and the current development efforts are focused in Shlomi Fish’s GitHub repo.

So, Shlomi Fish and other awesome contributors present to you, a new stable release (the last one was in distant 2009!) of PySolFC, new features:

[Fill in the features and improvements, shlomif, please]

And here you can find download links for current Windows, Linux, Android version [Err… I don’t think we have a new macOS version yet?, shlomif. Any other operating systems? Would it work on BSD?]

Future plans:

— vector-based fully scalable cards and graphics

[And what else shlomif?]

If you want to help, we need:

— developers

— graphic designers who would help with improving the app’s look

— people to help with documentation and translations

— people who would spread the word

We will gladly answer any questions!

Remember that apart from this thread you can also contact us using the FOSS Card Games group and a real time Internet chat on the ##pysol chat room on Freenode (note the double octothorpe/hash-sign/pound-sign).

**Comment by Alucard648** [comment]
With my small screen (1024*768) 2.0 cardsets were fine for me.

**Comment by hairspring** [comment]
@jan-kleks While it would be good to have choice, don't imagine everyone with a full HD display wants larger cards.  My wife completely rejected the idea when I offered to do work on this for her.  She likes having a smaller window on her full HD screen because it means she doesn't have to move the mouse as far.

The only time she wanted rescaled cards was several years ago, a short period when she had to run on a 1024x768 display.  I made a SMALLER cardset to overcome the problem of a game going off the edge of the monitor!

@Alucard648  I guess it varies with different games, some may fit 1024x768, but some don't.

**Comment by jan-kleks** [comment]
@hairspring Hopefully, proper vector support would fix all these problems with different resolutions (small and large). Okay then, so maybe not as a default cardset as I suggested in my original post, but I still think it would be nice to ship at least one cardset for people who would like to play with big cards in full HD, as a temporary measure. I have already suggested to @shlomif that this could be achieved by simply resizing our default cards (because of their simple pixelated look, they should scale nicely).

BTW, welcome back hairspring. :)

**Comment by hairspring** [comment]
@jan-kleks Yes, vector support would be great, I think I suggested it years ago on SourceForge... but I'm good at suggesting things without implementing them! :(

> nice to ship at least...

Absolutely.  Choice is good! :)

> BTW, welcome back hairspring. :)

[blush] Wow, you remember me!  I feel all warm and hugged! :)  Thanks.

I've been very remiss, have been watching from the wings all this time, but priorities haven't yet allowed me to contribute.  To explain... one day my wife said "Did you know you have 48 PCs around the house?".  Admittedly I do tend to hoard old items that need repair and I *loathe* those that just take stuff to the council tip.

But 48?  Clearly she was exaggerating.  I did my own count.  And returned, somewhat sheepishly, to admit that, though her figure was wrong, it was a little over 50!  So instead of programming, such spare time as I've had has been occupied refurb'ing and selling on ebay and ebid.net.  The task continues.

I've been very happy to see the progress on PySol, this group certainly has attitudes and ideals that I admire.  E.g. this week it's pleasing to see that both Python 2 and 3 are still in the frame, and I loved Shlomif's link to Joel's article -- just because Z works, that doesn't invalidate the need for ctrl+Z, the principle of least surprise.

I'll be back! :)

**Comment by rexx790** [comment]
Hi Jan
Totally agree that larger cardsets are a must for modern day computers.
I use a 32" monitor and a resolution of 1920x1080. This is a common setup these days and the small cards are a joke. I am the creator of the cardset you have linked in your opening post, and having full screen cards make pysol look better than any other card game out there. I can't help with coding or vector support but i can help with cards if needed.

**Comment by cardset** [comment]
In my opinion, the first cardset when pysol opens should be impressive or many people close the game and go away, looking for another game with nicer gfx. 

For example the dondorf set, from pysolfc.
Nice, historical, but small and no rankings on it.
And  very hard to play with it.
![grafik](https://user-images.githubusercontent.com/48878078/77225192-4a471080-6b6d-11ea-9ba1-1f0339439777.png)

and here is a dondorf set i use
![grafik](https://user-images.githubusercontent.com/48878078/77225286-01dc2280-6b6e-11ea-80b9-603f853aa457.png)

or if you want a bigger one:
![grafik](https://user-images.githubusercontent.com/48878078/77225302-26d09580-6b6e-11ea-87a6-fc975ff6039d.png)

**Comment by joeraz** [comment]
Regarding hi-res cardsets, this is something I am actively working on improving support for.  The next release (which I'm working on preparing) will have a number of features that enhance the display when using higher resolution cardsets, including adding a Hi-Res Cardsets category.  The first version of the Hi-Res Beta Cardsets package, containing the high resolution "Neo" cardset is already available to download on SourceForge: https://sourceforge.net/projects/pysolfc/files/PySolFC-Cardsets/Hi-Res%20Beta/PySolFC-Cardsets-HiRes-Beta-0.1/

Here's a screenshot showing the updates in action in Windows:
![HiRes](https://user-images.githubusercontent.com/2691813/125174018-5c98f380-e190-11eb-84d5-b64e7cc7a15a.png)

I plan to make this cardset the default eventually.  However, I'm not yet ready to do so for two main reasons:
- Currently, I only have the Neo cardset, which is a French type.  I'm working on building a Neo Matrix and can probably piece together a Neo Tarock or Hanafuda cardset from public domain pictures from the internet.  However, some of the remaining cardset types are pretty obscure.  I'd like to have at least one high resolution cardset of each type.
- Higher resolution cardsets are dependent on Auto-Scale being enabled to display properly.  There are still a few games that have images or text overlapping when Auto-Scale is used, especially when Auto-Scale is reducing the cardset size.  I've been able to fix a few of them, but can't say I've been able to check or fix every single game.  And similarly, I want to see if there are any other features that may have other display issues as a result of using higher resolution cardsets.  I want to give this a few releases so this can be tested more extensively - dated cardsets may be an issue, but a broken display would be even worse.

Since this is one of the most commented on issues with PySol, I figured it was worth a little mention of what I'm up to.  If anyone's interested in helping out with this, let me know.
